### ISSUE TITLE
Installation on macOS fails

### ISSUE THREAD

**Issue by Alok** [ray-project/ray#1953]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS 10.13.4
- **Ray installed from (source or binary)**: source
- **Ray version**: (latest build from master)
- **Python version**: 3.6.4, Anaconda
- **Exact command to reproduce**: 
  - Create conda env. 
  - `source activate ray
  - `pip install -e . --verbose` from `ray/python` dir
  - [get this error log](https://gist.github.com/cbbf4aad8208dfa358c1f62bfa1339b8)

### Describe the problem

I've got an RLlib implementation of TRPO for PyTorch that I want to add. I tried building ray from source along with my new implementation, but building fails.

I'm running in a conda env, and followed the instructions [here](https://ray.readthedocs.io/en/latest/installation.html#building-ray-from-source).

The issue seems to be with Parquet not being found.

### Source code / logs

https://gist.github.com/cbbf4aad8208dfa358c1f62bfa1339b8

**Comment by robertnishihara** [comment]
Can you try running (from `ray/`)

```
rm -rf python/ray/core
rm -rf python/ray/pyarrow_files
rm -rf thirdparty/build/arrow
rm -rf thirdparty/pkg/arrow
rm -rf thirdparty/build/parquet-cpp

git checkout python/ray/core
git checkout python/ray/pyarrow_files
```

and then rebuilding? There's some state that gets carried over so rebuilding after doing a git pull often doesn't work.

**Comment by Alok** [comment]
Worked.

**Comment by guoyuhong** [comment]
The same problem bothered me until I found this page...Thanks!

**Comment by naifrec** [comment]
this also works for me, thank you very much!

Additional to this issue, I also had another issue with an error of the sort being thrown:

```bash
      File "/Users/gsautie/code/soc/ppm/venv/lib/python3.6/re.py", line 142, in <module>
        class RegexFlag(enum.IntFlag):
    AttributeError: module 'enum' has no attribute 'IntFlag'
    make[2]: *** [CMakeFiles/_plasma_pyx] Error 1
    make[1]: *** [CMakeFiles/_plasma_pyx.dir/all] Error 2
    make: *** [all] Error 2
    error: command 'make' failed with exit status 2
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/Users/gsautie/code/libraries/ray/python/setup.py", line 155, in <module>
        license="Apache 2.0")
      File "/Users/gsautie/code/soc/ppm/venv/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
        return distutils.core.setup(**attrs)
      File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/Users/gsautie/code/soc/ppm/venv/lib/python3.6/site-packages/setuptools/command/develop.py", line 36, in run
        self.install_for_development()
      File "/Users/gsautie/code/soc/ppm/venv/lib/python3.6/site-packages/setuptools/command/develop.py", line 136, in install_for_development
        self.run_command('build_ext')
      File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/Users/gsautie/code/libraries/ray/python/setup.py", line 80, in run
        subprocess.check_call(["../build.sh", "-p", sys.executable])
      File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 291, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['../build.sh', '-p', '/Users/gsautie/code/soc/ppm/venv/bin/python3.6']' returned non-zero exit status 1.
  Can't roll back ray; was not uninstalled

```

It was solved by uninstalling `enum34`, as detailed in [this stack overflow post](https://stackoverflow.com/questions/43124775/why-python-3-6-1-throws-attributeerror-module-enum-has-no-attribute-intflag):

```bash
pip uninstall enum34
```

**Comment by dbieber** [comment]
@robertnishihara's comment https://github.com/ray-project/ray/issues/1953#issuecomment-384498836 was very helpful.

To get ray to build from source on macOS High Sierra, I also did:
- `pip uninstall enum34` as suggested by @naifrec (not sure if that was necessary)
- installed snappy (`brew install snappy`)
- installed thrift (`brew install thrift`)
- and installed jdk 10 (downloaded from oracle's website).

fwiw I was installing in a virtualenv with the command `pip install -e . --verbose`
I also was originally trying to install from source where one of the directories in the path to the source had a space in the name, but the install script broke on the space so I had to move the git repo to another location.
